src/test/java/org/mifos/integrationtest/cucumber/stepdef/ReportManagementSteps.java

```diff
@@ -0,0 +1,312 @@
+import org.mifos.integrationtest.config.OperationsAppConfig;
+import org.mifos.integrationtest.config.TenantConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+public class ReportManagementSteps extends BaseStepDef {
+
+    @Autowired
+    TenantConfig tenantConfig;
+    @Autowired
+    ScenarioScopeState scenarioScopeState;
+
+    Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    OperationsAppConfig operationsAppConfig;
+
+    @Given("Have tenant as {string}")
+    public void iHaveTenannt(String tenant) {
+        scenarioScopeState.tenant = tenantConfig.getTenant(tenant.toLowerCase());
+    }
+
+    @When("I call the get list of reports API with expected status of {int}")
+    public void iCallTheTransferAPIWithExpectedStatusOf(int expectedStatus) {
+        RequestSpecification requestSpec = Utils.getDefaultSpec(scenarioScopeState.tenant);
+        scenarioScopeState.response = RestAssured.given(requestSpec).baseUri(operationsAppConfig.operationAppContactPoint).expect()
+                .spec(new ResponseSpecBuilder().expectStatusCode(expectedStatus).build()).when().get(operationsAppConfig.reportEndpoint)
+                .andReturn().asString();
+
+        logger.info("ReportGet Response: {}", scenarioScopeState.response);
+    }
+
+    @Then("the response should contain a list of reports")
+    public void theResponseShouldContainListOfReports() {
+        Assert.assertNotNull(scenarioScopeState.response);
+    }
+
+    @When("I call the create report API with valid data with expected status of {int}")
+    public void iCallCreateReportAPIWithValidData(int expectedStatus) {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        ReportRequestDTO reportDTO = new ReportRequestDTO();
+
+        reportDTO.setReportName("Sample Report");
+        reportDTO.setReportType("Financial");
+        reportDTO.setReportSubType("Monthly");
+        reportDTO.setReportCategory("Operational");
+        reportDTO.setDescription("Financial summary for the month of June");
+        reportDTO.setReportSql("SELECT * FROM transactions WHERE date >= '2024-06-01' AND date <= '2024-06-30'");
+        List<ReportParameter> reportParameters = new ArrayList<>();
+
+        ReportParameter parameter1 = new ReportParameter();
+        parameter1.setParameterKey("Param1");
+        parameter1.setParameterValue("Value1");
+
+        ReportParameter parameter2 = new ReportParameter();
+        parameter2.setParameterKey("Param2");
+        parameter2.setParameterValue("Value2");
+
+        reportParameters.add(parameter1);
+        reportParameters.add(parameter2);
+
+        reportDTO.setReportParameters(reportParameters);
+
+        try {
+            scenarioScopeState.createReportBody = objectMapper.writeValueAsString(reportDTO);
+        } catch (JsonProcessingException e) {
+            logger.error("Unable to convert the DTO : {}", e);
```

Comment:
The test should fail here as the request body is not created. What are you sending in your API request by continuing the execution?